react/RedirectList.tsx

```diff
@@ -172,44 +152,144 @@ class RedirectList extends Component<Props, State> {
           fetchMoreResult
             ? {
                 ...prevData,
-                redirects: {
-                  ...prevData.redirects,
-                  redirects: [
-                    ...prevData.redirects.redirects,
-                    ...fetchMoreResult.redirects.redirects,
+                redirect: {
+                  ...prevData.redirect,
+                  list: [
+                    ...prevData.redirect.list,
+                    ...fetchMoreResult.redirect.list,
                   ],
                 },
               }
             : prevData,
         variables: {
-          from: this.state.paginationTo,
+          from: paginationTo,
           to: nextPaginationTo,
         },
       })
     }
 
-    this.setState(prevState => ({
-      ...prevState,
+    setPagination(prevState => ({
       paginationFrom: prevState.paginationTo,
       paginationTo: nextPaginationTo,
     }))
   }
 
-  private getNextPaginationTo = (paginationFrom: number, total: number) =>
-    total > paginationFrom + PAGINATION_STEP
-      ? paginationFrom + PAGINATION_STEP
-      : total
-
-  private handlePrevPageNavigation = () => {
-    this.setState(prevState => ({
-      ...prevState,
+  const handlePrevPageNavigation = useCallback(() => {
+    setPagination(prevState => ({
       paginationFrom: prevState.paginationFrom - PAGINATION_STEP,
       paginationTo: prevState.paginationFrom,
     }))
-  }
+  }, [setPagination])
+
+  const [isImportErrorModalOpen, setIsImportErrorModalOpen] = useState(false)
 
-  private openModal = () => this.setState({ isModalOpen: true })
-  private handleModalClose = () => this.setState({ isModalOpen: false })
+  return (
+    <>
+      <Helmet>
+        <title>{intl.formatMessage(messages.title)}</title>
+      </Helmet>
+      <Query<RedirectListQueryResult, RedirectListVariables>
+        notifyOnNetworkStatusChange
+        query={Redirects}
+        variables={{
+          from: PAGINATION_START,
+          to: PAGINATION_STEP,
+        }}
+      >
+        {({ data, fetchMore, loading, refetch, error }) => {
+          const redirects = (data && data.redirect && data.redirect.list) || []
+          const total =
+            (data && data.redirect && data.redirect.numberOfEntries) || 0
+          const hasRedirects = redirects.length > 0
+
+          return (
+            <>
+              {loading ? (
+                <Loader />
+              ) : !error ? (
```

Comment:
⛏ I believe it'd be cleaner to use a couple of separate `if`s and `return`s for the loading and error cases.